qiskit/pulse/schedule.py

```diff
@@ -309,7 +319,10 @@ def append(self, schedule: ScheduleComponent,
 
     def flatten(self) -> 'Schedule':
         """Return a new schedule which is the flattened schedule contained all ``instructions``."""
-        return Schedule(*self.instructions, name=self.name)
+        schedule = Schedule(name=self.name)
+        for instruction in self.instructions:
+            schedule.insert(*instruction, inplace=True)
```

Comment:
Considering the constructor is being deprecated, should a new method that insert a list of schedules be added, to avoid having to use this pattern in several places? I counted 3 occurrences in this file, and at least that many (probably more) in the rest of the library.

Reply:
I believe this is better suited by a for loop:
```python
for t0, inst in instructions:
     self.insert(t0, instruction, inplace=True)
```
Outside of this class, the functionality is not often used.

Reply:
What do you think is best? It's no bother for me to add the new method, but if it's not necessary/valuable, we can leave it out, of course.